File: LeetCode75_LevelTwo/Multiply_Strings/Multiply_Strings.py
# 43. Multiply Strings Python

class Solution(object):
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        if num1 == "0" or num2 == "0":
            return "0"
        num = 0
        n1 = 0
        for i in range(len(num1)):
            n1 += int(num1[i]) * pow(10, len(num1)-1-i)
        for j in range(len(num2)):
            n2 = int(num2[j]) * pow(10, len(num2)-1-j)
            num += n1 * n2
        return str(num)

File: LeetCode75_LevelTwo/Multiply_Strings/Multiply_Strings.cc
// 43. Multiply Strings C++

#include <string>
using namespace std;

class Solution {
public:
    string multiply(string num1, string num2) {
        int length1 = num1.size();
        int length2 = num2.size();
        if (length1 == 0 || length2 == 0) return "0";
        string ans(length1 + length2, '0');
        for (int i = length1 - 1; i >= 0; i--) {
            for (int j = length2 - 1; j >= 0; j--) {
                int sum = (num1[i] - '0') * (num2[j] - '0') + (ans[i + j + 1] - '0');
                ans[i + j + 1] = sum % 10 + '0';
                ans[i + j] += sum / 10;
            }
        }
        for (int i = 0; i < length1 + length2; i++){
            if (ans[i] != '0') {
                return ans.substr(i);
            }
        }
        return "0";
    }
};
